docs/DeveloperGuide.md

```diff
@@ -36,76 +36,93 @@ The high-level design of our program is based on a 3-tier architecture which con
-use of the package java.util.ArrayList. The TopicList is used to store Topics. Additionally,
-each topic stores a NoteList, which contains Notes.
+`TopicList` is an ArrayList of type `Displayable`, which is 1 of the 3 interfaces implemented 
+in the code for EDuke8. As such, many of the commands that manipulate the `TopicList` make 
+use of the package `java.util.ArrayList`. 
 
-1. Listing topics in TopicList
-2. Adding a new note
-3. Deleting an existing note
-4. Listing out all notes in a topic
+The `TopicList` is used to store `Topics`. `Topics` themselves implement the interface `Displayable`. 
+
+![TopicList](images/TopicListClassDiagram.png)
+
+There is just 1 command that manipulates the `TopicList`, which is `topics`. The `topics` command shows all the 
+`Topic` objects in the current `TopicList`.
 
-![TopicList_Class_Diagram](./images/TopicListAndNotes.png)
 
 #### 2.2.2. Implementation of TopicList
 
 **Listing topics in TopicList:**
 
-![TopicListSampleSequence](./images/TopicListSampleSequence.png)
-
 This task is performed by the `TopicList.showTopics()` method.
 
-Step 1. The `parseCommand()` method instantiates a `TopicsCommand` object which then calls the 
+![topicsSequence](images/topiclistsequence.png)
+
+Step 1: The `parseCommand()` method instantiates a `TopicsCommand` object which then calls the 
         `TopicList.showTopics()` method.
         
 Step 2. The `TopicList.showTopics()` method then calls the method `Ui.printTopicList()`. The 
         current `TopicList` is passed into the called method.
-        
-Step 3. The `Ui.printTopicList()` method then prints out the description of each topic in the 
-        `TopicList`. 
 
-`NoteList` is also an `ArrayList` of type `Displayable`, which is one of two interfaces implemented in the code for 
-E-Duke-8. As such, many of the commands that manipulate the `TopicList` make use of the package `java.util.ArrayList`. 
-The `NoteList` stores `Note` objects. Each topic has 1 `NoteList`. 
+Step 3: The `Ui.printTopicList()` method then prints out the number of questions in each topic,
+        along with the description of each topic in the `TopicList`. 
+
+
+#### 2.2.3. Design of NoteList
+
+`NoteList` is also an `ArrayList` of type `Displayable`. As such, just like with `TopicList`, many of the commands 
+that manipulate the `NoteList` also make use of the package `java.util.ArrayList`. 
+
+Each topic has an attribute of type `NoteList`, which contains `Notes`.
+
+![NoteListClass](images/NoteListClassDiagram.png)
 
-#### 2.2.3. Implementation of Notes
+#### 2.2.4. Implementation of Notes
+
+There are 3 commands that involve the manipulation of `NoteList`. They are: `note add`, which adds a new note, 
+`note delete`, which deletes a note, and `note list`, which lists out all `Note` objects in a `NoteList` from a 
+specified `Topic`. These commands are carried out by the instantiation of a `NoteCommand` object. The `NoteCommand` 
```

Comment:
Maybe it would be better to refer to them as objects  as opposed to their abstract meaning? For example instead of " `Note` objects in a `NoteList` from a specified `Topic`. " you can say "`Note` objects stored in a `NoteList` object that share the same `Topic` parameter."

Reply:
Thanks for your suggestion! Will do